nfs/nfs.go

```diff
@@ -0,0 +1,98 @@
+package nfs
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/longhorn/go-common-libs/types"
+)
+
+const (
+	nfsmountGlobalSection  = "NFSMount_Global_Options"
+	nfsmountDefaultVersion = "Defaultvers"
+)
+
+var ErrNotConfigured = errors.New("is not configured")
+
+// GetSystemDefaultNFSVersion reads the system default NFS version. This config can be overridden by nfsmount.conf under
+// configDir. If configDir is empty, it will be /etc by default. If the nfsmount.conf is absent, or the default
+// version is not overridden in the configuration file, an ErrNotConfigured error will be returned.
+func GetSystemDefaultNFSVersion(configDir string) (major, minor int, err error) {
+	if configDir == "" {
+		configDir = types.SysEtcDirectory
+	}
+
+	configMap, err := getSystemNFSMountConfigMap(configDir)
+	if os.IsNotExist(err) {
+		return 0, 0, ErrNotConfigured
+	} else if err != nil {
+		return 0, 0, err
+	}
+	if globalSection, ok := configMap[nfsmountGlobalSection]; ok {
+		if configured, ok := globalSection[nfsmountDefaultVersion]; ok {
+			majorStr, minorStr, _ := strings.Cut(configured, ".")
+			major, err := strconv.Atoi(majorStr)
+			if err != nil {
+				return 0, 0, errors.Wrapf(err, "invalid NFS major version %q", configured)
+			}
+
+			minor := 0
+			if minorStr != "" {
+				minor, err = strconv.Atoi(minorStr)
+				if err != nil {
+					return 0, 0, errors.Wrapf(err, "invalid NFS minor version %q", configured)
+				}
+			}
+			return major, minor, nil
+		}
+	}
+	return 0, 0, ErrNotConfigured
```

Comment:
Suggest to avoid common error messages. Instead, we should use a more descriptive message that makes debugging easier. For example:
```suggestion
	return 0, 0, fmt.Errorf("cannot retrieve system default NFS version from %q", configDir)
```

Reply:
Maybe we can wrap the common error with a meaningful message, so that it would be possible for the caller to tell if the configuration is overridden or not.
```suggestion
	return 0, 0, errors.Wrapf(ErrNotConfigured, "system default NFS version is not overridden under %q", configDir)
```

Reply:
Sounds good to me. Thank you!

Reply:
> Could you explain the benefit of taking this approach (wrapping with a common error)?

After talking to @COLDTURNIP , this is more of an error type. The benefit of having a common error type is that it allows the callers to consistently handle a specific error condition, regardless of differences in the error messages.

Reply:
The common error `ErrNotConfigured` is now moved to `types.ErrNotConfigured`

Reply:
And there's an example for unwrapping the error in `nfs/nfs_test.go`.